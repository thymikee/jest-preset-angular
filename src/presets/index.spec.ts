import fs from 'fs';
import path from 'path';

import { defaultPreset, defaultEsmPreset } from './';

describe('Jest presets', () => {
  test.each([defaultPreset, defaultEsmPreset])('should return the correct jest config', (preset) => {
    expect(preset).toMatchSnapshot();
  });

  test('should have the correct types which come from `ts-jest`', () => {
    expect(fs.readFileSync(path.join(__dirname, '..', '..', 'presets/index.d.ts'), 'utf-8')).toMatchSnapshot();
  });
});
