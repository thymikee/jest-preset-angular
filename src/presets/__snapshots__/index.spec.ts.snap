// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest presets should have the correct types which come from \`ts-jest\` 1`] = `
"declare const _default: {
  defaults: {
    transformIgnorePatterns: string[];
    transform: import('ts-jest').JestConfigWithTsJest['transform'];
    testEnvironment: string;
    moduleFileExtensions: string[];
    snapshotSerializers: string[];
  };
  defaultsESM: {
    extensionsToTreatAsEsm: string[];
    moduleNameMapper: {
      tslib: string;
    };
    transform: import('ts-jest').JestConfigWithTsJest['transform'];
    transformIgnorePatterns: string[];
    testEnvironment: string;
    moduleFileExtensions: string[];
    snapshotSerializers: string[];
  };
  defaultTransformerOptions: import('ts-jest').TsJestTransformerOptions;
};
export default _default;
"
`;

exports[`Jest presets should return the correct jest config 1`] = `
{
  "moduleFileExtensions": [
    "ts",
    "html",
    "js",
    "json",
    "mjs",
  ],
  "snapshotSerializers": [
    "jest-preset-angular/build/serializers/html-comment",
    "jest-preset-angular/build/serializers/ng-snapshot",
    "jest-preset-angular/build/serializers/no-ng-attributes",
  ],
  "testEnvironment": "jsdom",
  "transform": {
    "^.+\\.(ts|js|mjs|html|svg)$": [
      "jest-preset-angular",
      {
        "stringifyContentPathRegex": "\\.(html|svg)$",
        "tsconfig": "<rootDir>/tsconfig.spec.json",
      },
    ],
  },
  "transformIgnorePatterns": [
    "node_modules/(?!.*\\.mjs$)",
  ],
}
`;

exports[`Jest presets should return the correct jest config 2`] = `
{
  "extensionsToTreatAsEsm": [
    ".ts",
  ],
  "moduleFileExtensions": [
    "ts",
    "html",
    "js",
    "json",
    "mjs",
  ],
  "moduleNameMapper": {
    "tslib": "tslib/tslib.es6.js",
  },
  "snapshotSerializers": [
    "jest-preset-angular/build/serializers/html-comment",
    "jest-preset-angular/build/serializers/ng-snapshot",
    "jest-preset-angular/build/serializers/no-ng-attributes",
  ],
  "testEnvironment": "jsdom",
  "transform": {
    "^.+\\.(ts|js|html|svg)$": [
      "jest-preset-angular",
      {
        "stringifyContentPathRegex": "\\.(html|svg)$",
        "tsconfig": "<rootDir>/tsconfig.spec.json",
        "useESM": true,
      },
    ],
  },
  "transformIgnorePatterns": [
    "node_modules/(?!tslib)",
  ],
}
`;
